sashimi-webapp/src/components/file-manager/FileManager.vue

```diff
@@ -16,10 +24,102 @@ export default {
   },
   data() {
     return {
-      action: '',
+      viewMode: 'listView',
+      docs: {},
+      history: null
     };
   },
   watch: {
+  },
+  methods: {
+    changeFolder(newFolder) {
+      if (newFolder.id === 0) {
+        this.$router.push({ path: '' });
+      } else {
+        this.$router.push({ path: '', query: { folder: newFolder.id } });
+      }
+      this.docs = newFolder;
+      this.history.update(this.docs);
+    },
+    changeViewMode(viewMode) {
+      this.viewMode = viewMode;
+    },
+    executeAction(action, doc) {
```

Comment:
Definitely need to refactor this part next time by grouping the relevant conditions together.

e.g. [Creation]. [History], [Operation] etc